kasmvnc/run.sh

```diff
@@ -14,166 +15,169 @@ check_installed() {
+
+  if command -v curl &> /dev/null; then
+    download_tool="curl -fsSL"
+  elif command -v wget &> /dev/null; then
+    download_tool="wget -q -O-"
   elif command -v busybox &> /dev/null; then
-    busybox wget -O $output $url
+    download_tool="busybox wget -O-"
   else
-    echo "Neither wget, curl, nor busybox is installed. Please install one of them to proceed."
+    echo "ERROR: No download tool available (curl, wget, or busybox required)"
     exit 1
   fi
-}
 
-# Function to install kasmvncserver for debian-based distros
-install_deb() {
-  local url=$1
-  download_file $url /tmp/kasmvncserver.deb
-  sudo apt-get update
-  DEBIAN_FRONTEND=noninteractive sudo apt-get install --yes -qq --no-install-recommends --no-install-suggests /tmp/kasmvncserver.deb
-  sudo adduser $USER ssl-cert
-  rm /tmp/kasmvncserver.deb
+  $download_tool "$url" > "$output" || {
+    echo "ERROR: Failed to download $url"
+    exit 1
+  }
 }
 
-# Function to install kasmvncserver for Oracle 8
-install_rpm_oracle8() {
-  local url=$1
-  download_file $url /tmp/kasmvncserver.rpm
-  sudo dnf config-manager --set-enabled ol8_codeready_builder
-  sudo dnf install oracle-epel-release-el8 -y
-  sudo dnf localinstall /tmp/kasmvncserver.rpm -y
-  sudo usermod -aG kasmvnc-cert $USER
-  rm /tmp/kasmvncserver.rpm
+# Add user to group using available commands
+add_user_to_group() {
+  local user="$1"
+  local group="$2"
+
+  if command -v usermod &> /dev/null; then
+    sudo usermod -aG "$group" "$user"
+  elif command -v adduser &> /dev/null; then
+    sudo adduser "$user" "$group"
+  else
+    echo "ERROR: At least one of 'adduser'(Debian) 'usermod'(RHEL) is required"
+    exit 1
+  fi
 }
 
-# Function to install kasmvncserver for CentOS 7
-install_rpm_centos7() {
+# Function to install kasmvncserver for debian-based distros
+install_deb() {
   local url=$1
-  download_file $url /tmp/kasmvncserver.rpm
-  sudo yum install epel-release -y
-  sudo yum install /tmp/kasmvncserver.rpm -y
-  sudo usermod -aG kasmvnc-cert $USER
-  rm /tmp/kasmvncserver.rpm
+  download_file "$url" /tmp/kasmvncserver.deb
+  # Define the directory to check
+  CACHE_DIR="/var/lib/apt/lists/partial"
+  # Check if the directory exists and was modified in the last 60 minutes
+  if [ ! -d "$CACHE_DIR" ] || ! find "$CACHE_DIR" -mmin -60 -print -quit &> /dev/null; then
+    echo "Stale Package Cache, updating..."
+    # Update package cache with a 300-second timeout for dpkg lock
+    sudo apt-get -o DPkg::Lock::Timeout=300 -qq update
+  fi
+  DEBIAN_FRONTEND=noninteractive sudo apt-get -o DPkg::Lock::Timeout=300 install --yes -qq --no-install-recommends --no-install-suggests /tmp/kasmvncserver.deb
+  add_user_to_group "$USER" ssl-cert
+  rm /tmp/kasmvncserver.deb
 }
 
 # Function to install kasmvncserver for rpm-based distros
 install_rpm() {
   local url=$1
-  download_file $url /tmp/kasmvncserver.rpm
+  download_file "$url" /tmp/kasmvncserver.rpm
   sudo rpm -i /tmp/kasmvncserver.rpm
```

Comment:
We should look into finding or building a desktop image to test this with for RHEL, Fedora, Centos, and Oracle.
As with the original script I believe only the `codercom/enterprise-desktop:ubuntu` image was tested and this is the one I am using.

I did some further reading, it sounds like on RHEL8, [yum is an alias for dnf](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/8/html/8.4_release_notes/distribution-of-content-in-rhel-8#package_management_with_yum_dnf).

> On Red Hat Enterprise Linux 8, installing software is ensured by the YUM tool, which is based on the DNF technology. We deliberately adhere to usage of the yum term for consistency with previous major versions of RHEL. However, if you type dnf instead of yum, the command works as expected because yum is an alias to dnf for compatibility.

Other places suggest that dnf is preferred almost always to yum for the extra features.

I've also read that we should use `sudo dnf localinstall /tmp/kasmvncserver.rpm` instead.